src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/EnvelopeProcessor.java

```diff
@@ -6,28 +6,35 @@
 import org.springframework.transaction.annotation.Transactional;
 import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
 import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeRepository;
-import uk.gov.hmcts.reform.bulkscanprocessor.entity.ScannableItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeStatusRepository;
 import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.MetadataNotFoundException;
 import uk.gov.hmcts.reform.bulkscanprocessor.util.EntityParser;
+import uk.gov.hmcts.reform.bulkscanprocessor.util.EnvelopeStatusBuilder;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.util.List;
 import java.util.Objects;
 
+import static uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeStatusEnum.DOC_UPLOADED;
+
 @Component
 public class EnvelopeProcessor {
     private static final Logger log = LoggerFactory.getLogger(EnvelopeProcessor.class);
 
     private final EnvelopeRepository envelopeRepository;
+    private final EnvelopeStatusRepository envelopeStatusRepository;
 
-    public EnvelopeProcessor(EnvelopeRepository envelopeRepository) {
+    public EnvelopeProcessor(
+        EnvelopeRepository envelopeRepository,
+        EnvelopeStatusRepository envelopeStatusRepository
+    ) {
         this.envelopeRepository = envelopeRepository;
+        this.envelopeStatusRepository = envelopeStatusRepository;
     }
 
     @Transactional
-    public List<ScannableItem> processEnvelope(byte[] metadataStream) throws IOException {
+    public Envelope processEnvelope(byte[] metadataStream) throws IOException {
```

Comment:
Since you are now returning the envelope and not retrieving scannable I think you don't need transactional annotation here ?